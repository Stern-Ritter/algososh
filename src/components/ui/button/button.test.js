import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Component Button", () => {
  it("renders button with text", () => {
    const tree = renderer.create(<Button text="Test button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders disabled button", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders button with the loading indicator", () => {
    const tree = renderer.create(<Button isLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls handler on click", () => {
    const mockHandler = jest.fn();

    render(<Button text="Test button" onClick={mockHandler} />);
    const button = screen.getByText("Test button");

    fireEvent.click(button);
    expect(mockHandler).toHaveBeenCalledTimes(1);
  });
});
