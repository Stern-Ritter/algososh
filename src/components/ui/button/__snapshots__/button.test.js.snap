// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button renders button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test button
  </p>
</button>
`;

exports[`Component Button renders button with the loading indicator 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Component Button renders button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Component Button renders disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
